docs/learn/calls.md

```diff
@@ -174,8 +183,36 @@ response = recommend_book("fantasy")
 print(response.content)
 ```
 
+In this example, we're creating a custom OpenAI client with a specific API key, organization ID, and base URL. This custom client is then passed to the call decorator, allowing us to use a custom endpoint or configuration for our LLM calls.
+
 Any custom client is supported so long as it has the same API as the original base client.
 
+## Using the `run` Method
+
+As an alternative to the functional prompt template approach, you can use the `run` method to execute your prompts with different providers. This method allows for more flexibility and easier provider switching. Here's an example:
+
+```python hl_lines="8 12"
+from mirascope.core import anthropic, openai, prompt_template
+
+@prompt_template("Recommend a {genre} book")
+def recommend_book(genre: str):
+    ...
+
+# Using OpenAI
+openai_response = recommend_book.run(openai.call(model="gpt-4o-mini"))("fantasy")
+print("OpenAI recommendation:", openai_response.content)
+
+# Using Anthropic
+anthropic_response = recommend_book.run(anthropic.call(model="claude-3-5-sonnet-20240620"))("fantasy")
+print("Anthropic recommendation:", anthropic_response.content)
+```
+
+This approach allows you to easily switch between providers or compare outputs without changing your core prompt logic.
+
+## Metadata
+
+You can add metadata to your calls using the `@metadata` decorator. This can be useful for tracking, categorizing, or adding any additional information to your LLM calls. For more information on using metadata, see the [Metadata documentation](./metadata.md).
```

Comment:
what are your thoughts on deprecating the `@metadata` decorator and instead only supporting metadata through dynamic configuration?

i feel like forcing this single path will be a little more clear

Reply:
Looks good.
I will remove the section and I will change the link to dynamic configuration markdown.